src/mrinufft/trajectories/tools.py

```diff
@@ -700,3 +701,178 @@ def radialize_center(trajectory, nb_samples, in_out=False):
+            "Edges will not be sampled. "
+        )
+    rng = np.random.default_rng(rng)  # validate get a RNG from a seed or existing rng.
+
+    def _get_samples(p) -> list[int]:
+        p /= np.sum(p)
+        return list(rng.choice(borders, size=n_samples_borders, replace=False, p=p))
+
+    if pdf == VDSpdf.GAUSSIAN:
+        p = norm.pdf(np.linspace(norm.ppf(0.001), norm.ppf(0.999), len(borders)))
+        sampled_in_border = _get_samples(p)
+    elif pdf == VDSpdf.UNIFORM:
+        p = np.ones(len(borders))
+        sampled_in_border = _get_samples(p)
+    elif pdf == VDSpdf.EQUISPACED:
+        sampled_in_border = list(borders[::accel])
+
+    elif isinstance(pdf, np.ndarray):
+        if len(pdf) == dim_size:
+            # extract the borders
+            p = pdf[borders]
+        elif len(pdf) == len(borders):
+            p = pdf
+        else:
+            raise ValueError("Invalid size for probability.")
+        sampled_in_border = _get_samples(p)
+
+    else:
+        raise ValueError("Unsupported value for pdf use any of . ")
+        # TODO: allow custom pdf as argument (vector or function.)
+
+    line_locs = np.array(sorted(center_indexes + sampled_in_border))
+    # apply order of lines
+    if order == VDSorder.CENTER_OUT:
+        line_locs = _flip2center(sorted(line_locs), dim_size // 2)
+    elif order == VDSorder.RANDOM:
+        line_locs = rng.permutation(line_locs)
+    elif order == VDSorder.TOP_DOWN:
+        line_locs = np.array(sorted(line_locs))
+    else:
+        raise ValueError(f"Unknown direction '{order}'.")
+    return (line_locs / dim_size) - 0.5  # rescale to [-0.5,0.5]
+
+
+def get_random_loc_nd(N_samples, shape, pdf):
+    """Get random location in N-dimensions."""
+    # TODO
+    pass
+
+
+def stack_random(
+    traj,
+    dim_size,
+    center_prop,
+    accel=4,
+    pdf="uniform",
+    rng=None,
+    order="center-out",
+):
+    """Stack a 2D trajectory with random location.
+
+    Parameters
+    ----------
+    traj: np.ndarray
+        Existing 2D trajectory.
+    dim_size: int
+        Size of the k_z dimension
+    center_prop: int or float
+        Number of line or proportion of slice to sample in the center of the k-space
+    pdf: str or np.array
+        Probability density function for the remaining samples.
+        "uniform" (default), "gaussian" or np.array
+    rng: random state
+    order: str
+        Order of the lines, "center-out" (default), "random" or "top-down"
+
+    Returns
+    -------
+    numpy.ndarray
+        The 3D trajectory stacked along the :math:`k_z` axis.
+
+    """
+    line_locs = get_random_loc_1d(dim_size, center_prop, accel, pdf, rng, order)
+    if len(traj.shape) == 2:
+        Nc, Ns = 1, traj.shape[0]
+    else:
+        Nc, Ns = traj.shape[:2]
+
+    new_traj = np.zeros((len(line_locs), Nc, Ns, 3))
+    for i, loc in enumerate(line_locs):
+        new_traj[i, :, :, :2] = traj
+        new_traj[i, :, :, 2] = loc
```

Comment:
`tools.stack` and most other tools allow the input trajectory to also be 3D in case we want to stack "flat" trajectories with thickness as shown in the tools example in the documentation. Would it be possible to make that possible too for this function ?